### ISSUE TITLE
raster support until February meeting

### ISSUE THREAD

**Issue by rsbivand** [r-spatial/discuss#17]
This is a place to use to keep information stemming from the R-sig-geo thread starting [here](https://stat.ethz.ch/pipermail/r-sig-geo/2017-October/026091.html) collected.

A cleaned version of 2.5.8 (re-versioned [2.6.1](http://spatial.nhh.no/R/raster/raster_2.6-1.tar.gz)) that adds src/init.c and src/raster.h and fixes trivial other issues is on my server. I have not fixed the errors resulting from CMD check --run-dontrun; getting them working would kickstart enhancing coverage. It would be very helpful if unit testing was contributed, first as a separate package.

The current trunk on R-Forge (2.5.17) does [not](https://r-forge.r-project.org/R/?group_id=294&log=build_src&pkg=raster&flavor=patched) build, C++ errors.

I'll be writing to Robert and others directly about this.

**Comment by JodiNorris** [comment]
I apologize I'm not familiar with etiquette on github, but wanted to add this issue:

I too greatly appreciate raster, and have been forced to work in the second-to-last version of it for over a year now because the most recent version (2.5-8) produces incorrect results for the **approxna** command. The issue is described here 
https://stackoverflow.com/questions/45066250/approxna-producing-different-and-incorrect-results-in-stacks-of-large-rasters

It would be fantastic if this could be resolved - even if the old working version of the command could be used instead of the new problematic version. Raster is valuable enough to me that all my base R and all packages are frozen back to the time of the earlier 2.3-12 version of raster to make sure that everything stays compatible and works correctly.

Best regards

Jodi Norris

**Comment by rsbivand** [comment]
Jodi,

Thanks for your input. This will crucially depend on user contributions. We need a set of tests from a baseline acknowledged as correct by "authorities", so that deviations can be tracked. This implies that those with an interest in contributing to stability should first establish sets of tests of expected results, then find ways to merge these into a free-standing package to test the raster package. Then the raster maintainer will be able to draw conclusions from that. 

As I mentioned, the dontrun block should all actually run too, but be hopped-over to save time. It is a symptom of failure in what is known as coverage if the code isn't stress-tested against expected results on a regular basis. 

Could you consider contributing and documenting a test for the `approxna` case?

**Comment by sculpin99** [comment]
Hello,
I am a raster package user with some experience building R packages, though I am not a sophisticated developer. If there is any way I can help with maintenance of the raster package, I am willing to donate up to 4 hours of work time a month doing so. Without the raster package I would be like a carpenter without a hammer.

**Comment by JodiNorris** [comment]
Yes, I would be happy to try to document a test for the approxna case.  I've never done this before but I'm looking over an example description of the test process for R packages here http://r-pkgs.had.co.nz/tests.html

If you have any other suggestions of what I should read or use as a reference please pass em along.

Thanks!

**Comment by rsbivand** [comment]
Using the full testthat framework may be a goal in the future, but for now it may be sufficient to write a script (like those in examples) with the expected values commented out. Then add specific tests checking that the expected and observed output agree (maybe exactly, maybe within some given precision, maybe also checking attributes). Using an incremental approach permits the checking of the results, and the detection of whether a correct outcome is being rejected for spurious reasons (especially when equality is required but all.equal() with machine precision tolerance would be OK).

**Comment by rsbivand** [comment]
Please also see the reply by Robert Hijmans on [R-sig-geo](https://stat.ethz.ch/pipermail/r-sig-geo/2017-October/026114.html).

**Comment by JodiNorris** [comment]
Sounds good on both counts, I've got some tests already made that I use for my own work that I can expand to contribute to a larger effort; but I'll wait for now and see which way Robert Hijmans wants to go with it.

**Comment by tim-salabim** [comment]
@rsbivand can this be closed?
